### PR TITLE
Add Swift Package Manager Support

### PR DESCRIPTION
**Pull request by witekbobrowski**
Since Swift 5.3 is now officially released with Xcode 12 and it brings support for resources, I have made an attempt to bring SPM support to DeviceGuru since people were asking for it.

This PR adds:
- `Package.swift` file for SPM integration
- New, improved rules in `.gitignore`
- Renamed `Sources/` directory (previously `Source/`)
- New generated Xcode project file using `swift package generate-xcodeproj` command
- Updated `README.md` with SPM instructions

Generator script have been tested to check if generated files land in correct directory.

The only thing left to do once this PR is merged is to tag new release `6.0.7` (as mentioned in the README.md) so the SPM will correctly recognise the library.

Cheers!

### CODE REVIEW

**Comment by witekbobrowski** [comment]
Also, would love to have this PR be hacktoberfest-eligable so If you find it ready to be merged please add `hacktoberfest-accepted` topic to the Pull Request. Thanks!

File: Generator/main.swift [hunk 1]
```diff
@@ -89,7 +89,7 @@ func main() {
         valueDict?.removeValue(forKey: "enum")
         deviceList[hardwareKey] = valueDict
     }
-    let dirPath = "../Source/"
+    let dirPath = "../Sources/"
```
Comment (InderKumarRathore) [inline Generator/main.swift:92]: I think this will break the cocoapod, need to update that file as well. Have to fix that one. Will check it

**Comment by InderKumarRathore** [comment]
BTW it looks good

File: Generator/main.swift (hunk 1 above)
Comment (witekbobrowski) [reply Generator/main.swift:92]: You are totally right, sorry I missed that!

Comment (InderKumarRathore) [reply Generator/main.swift:92]: It's all right, i fixed it

**Pull request merged**